### PR TITLE
fix: fail open for missing nodeclaims in termination

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
Updates the node termination controller to fail open rather than closed when a NodeClaim can not be found for the node. This is an invariant that shouldn't be broken, but discovered that it's currently possible if the node joins the cluster in time for the same reconciliation of the lifecycle controller to perform launch and registration **and** it fails to persist the provider-id to the NodeClaim's status. This likely will never happen in a real cloudprovider, but does occur in KWOK simulation testing. 

**How was this change tested?**
`make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/node/termination/controller.go [hunk 1]
```diff
@@ -91,21 +91,19 @@ func (c *Controller) finalize(ctx context.Context, node *corev1.Node) (reconcile
 		return reconcile.Result{}, nil
```
Comment (jonathan-innis) [inline pkg/controllers/node/termination/controller.go:91]: Should we add some testing for the no NodeClaim and duplicate NodeClaim cases?

**Review by jonathan-innis — APPROVED** [review]

File: pkg/controllers/node/termination/controller.go (hunk 1 above)
Comment (jmdeal) [reply pkg/controllers/node/termination/controller.go:91]: Good call, done

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

File: pkg/controllers/node/termination/controller.go (hunk 1 above)
Comment (jonathan-innis) [reply pkg/controllers/node/termination/controller.go:91]: Completed.
